Core/include/Acts/Utilities/Delegate.hpp

```diff
@@ -65,11 +64,12 @@ class Delegate<R(Args...), H, O> {
       Delegate<R(Args...), holder_type, DelegateType::Owning>;
   using NonOwningDelegate =
       Delegate<R(Args...), holder_type, DelegateType::NonOwning>;
+
   template <typename T>
-  using isNoFunPtr =
-      std::enable_if_t<!std::is_convertible_v<std::decay_t<T>, function_type> &&
-                       !std::is_same_v<std::decay_t<T>, OwningDelegate> &&
-                       !std::is_same_v<std::decay_t<T>, NonOwningDelegate>>;
+  using isNoFunPtr = std::conjunction<
+      std::negation<std::is_convertible<std::decay_t<T>, function_type>>,
```

Comment:
Yes, because `isNoFunPtr` is a type, not a boolean value. So you can't actually negate it using `!`, you need to use `std::negation`. I wanted to make this a proper concept at some point, but I was having some trouble with it (don't remember what exactly) so I gave up on that for now.

Reply:
Actually, let me try again if I can conceptify this properly.

Reply:
Ah! I remember; this was because `isNoFunPtr` binds a lot of typedefs for the class it lives in, which you are allowed to do for old-school trait classes but you cannot define concepts inside of types, so the concept would need to have way more arguments.